src/peer_manager.rs

```diff
@@ -1528,6 +1544,14 @@ impl PeerManager {
         }
     }
 
+    /// Removes the given entry from the routing_table or peer_map, returns the removal details
+    pub fn purge_out_of_sync_peer(&mut self,
+                                  name: &XorName)
+                                  -> Result<RemovalDetails<XorName>, RoutingTableError> {
+        let _ = self.peer_map.remove_by_name(name);
```

Comment:
will have a check tomorrow whether we can do that without expanding that epic inline closure :)